morphic.js

```diff
@@ -7818,10 +7820,47 @@ MenuMorph.prototype.select = function (aMenuItem) {
     this.selection = aMenuItem;
 };
 
+MenuMorph.prototype.mouseLeave = function () {
+    if (this.parent instanceof MenuMorph) {
+        // this is a submenu so we vanish when lose focus
+        var mouses = this.world.hand.allMorphsAtPointer();
+        if (contains(mouses, this.parent)) {
+            var items = mouses.filter(function (morph)
+                                      {return morph instanceof MenuItemMorph});
+            if (items[0].doubleClickAction != this) {
+                this.destroy();
+            }
+        } else {
+            this.destroy();
+        }
+    }
+}
+            
+
 MenuMorph.prototype.destroy = function () {
+var myself = this;
+
     if (this.hasFocus) {
         this.world.keyboardReceiver = null;
     }
+
+    if ((this.parent instanceof MenuMorph) && this.world) {
+        // If I'm a submenu, un-highlight the item pointing to me.
```

Comment:
No, you're misunderstanding.  This is when _leaving_ the submenu, not when entering it.  If you just mouse away from the submenu and it disappears, the highlighting of its MenuItem should be undone too.

OTOH I'm not sure why this works; if you can't check object equality, how does "contains" work?  Or is the "this.world" that I don't understand somehow work around the comparison not working?  I'd be delighted if you could explain that!